### DISCUSSION TITLE
Add SHA256 output to Release tags

### DISCUSSION THREAD

**Discussion by Uzaaft** [ghostty-org/ghostty#4229] [Feature Requests, Ideas]
I think adding sha256 to the output binary/dmg would be beneficial for package managers that uses this, like brew.

**Comment by mitchellh** [comment]
What files do you need it for? The dmg itself? And how would you want it published?

**Reply by Uzaaft** [reply]
In brews case it’s the dmg itself. And output-ed to the release tag should be enough(I have both done that myself before and seen other people do it that way). The checksums ensures that people can verify its the original dmg.

**Reply by Uzaaft** [reply]
I can come up with an example of a repo tomorrow.

**Reply by Uzaaft** [reply]
Sorru about the delay on this. Life got hectic.

A way to do this is like how the repo `sh` does it, where they generate the sha256 for every binary and then add it to a file:
https://github.com/mvdan/sh/releases/tag/v3.10.0

Image from the sha256 file:
![image](https://github.com/user-attachments/assets/4a522d96-4f4b-46a8-9caf-1c48d6d7625c)

**Comment by romangoempire** [comment]
I believe it’s possible to automate the process of creating a pull request (PR) to the Homebrew repository. Specifically, for updating the file [Brew Cask File](https://github.com/Homebrew/homebrew-cask/blob/e6feb2a3a2f8dbefd60c41a5a960ffc49a9fd66e/Casks/g/ghostty.rb), I propose the following approach using a GitHub Action:

Proposed Workflow:
1.	Trigger: The action should run upon publishing a new release.
2.	Environment: Use an Ubuntu runner for the workflow.
3.	Forking: Automatically fork the Homebrew repository.
4.	SHA256 Token Generation: Generate the sha256 token from the appropriate file (likely the tar.gz file, but this should be verified).
5.	File Update: Edit the specified file and replace the existing key with the new one.
6.	Create PR: Use brew bottle (or a similar command. I did it once, but I dont remember) to finalize the update and open a pull request.

I assume that you’ll need to pass Git credentials to ensure the PR can be created successfully.

Does this cover everything, or am I missing any key steps?
I’m happy to test the workflow and provide the .yml configuration file for the GitHub Action.

**Reply by mitchellh** [reply]
The Ghostty project doesn't own distribution (besides the uploaded macOS dmg). I'm not interested in this. But @Uzaaft might be for a separate repo.

**Reply by Uzaaft** [reply]
This is overengineered and not needed.

**Reply by romangoempire** [reply]
> This is overengineered and not needed. 

So your idea is to do it manually?
I am new to the brew updates and still learning  ^^

**Reply by JakeWharton** [reply]
It's not over engineered, but brew will do a lot of those steps for you.

5 & 6 are accomplished with `brew bump-formula-pr`. For 3 you probably just want to run the action in a permanent fork and have it do a `brew update` first. And then for 1 you can run on a schedule (every hour, say) to look for a new version.

All in all it's about 20 total lines of action Yaml and you're done.

**Reply by romangoempire** [reply]
> All in all it's about 20 total lines of action Yaml and you're done.

This is what I thought too, but I guess my explanation and knowledge how to solve it was lacking. Thank you for clarifying!

**Reply by Uzaaft** [reply]
The upstream brew cask will auto update from now on, so it is over-engineered. :)

**Reply by JakeWharton** [reply]
Oh, nice. That just makes it redundant. I only maintain my own taps, so the engineering is still correct.
